robottelo/upgrade_utility.py

```diff
@@ -0,0 +1,165 @@
+            run('rm -rf {0}'.format(repo_path + other_rpm))
+            run('createrepo --update {0}'.format(repo_path))
+        else:
+            run('rm -rf {}'.format(repo_path))
+            run('mkdir {}'.format(repo_path))
+            run('wget {0} -P {1}'.format(rpm_name, repo_path))
+            # Renaming custom rpm to preRepoSync.rpm
+            run('createrepo --database {0}'.format(repo_path))
+
+    def host_status(self, client_container_name=None):
+        """ fetch the content host details.
+        :param: str client_container_name: The content host hostname
+        :return: nailgun.entity.host: host
+        """
+        host = entities.Host().search(
+            query={'search': '{0}'.format(client_container_name)})
+        return host
+
+    def host_location_update(self, client_container_name=None, loc=None):
+        """ Check the content host status (as package profile update task does take time to
+        upload) and update location.
+
+        :param: str client_container_name: The content host hostname
+        :param: str loc: Location
+        """
+        if len(self.host_status(client_container_name=client_container_name)) == 0:
+            wait_for(
+                lambda: len(self.host_status(client_container_name=client_container_name
+                                             )) > 0,
+                timeout=100,
+                delay=2,
+                logger=self.logger
+            )
+        host_loc = self.host_status(client_container_name=client_container_name)[0]
+        host_loc.location = loc
+        host_loc.update(['location'])
+
+    def publish_content_view(self, org=None, repolist=None):
+        """
+        publish content view and return content view
+        :param: str org: Name of the organisation
+        :param: str repolist: Name of the repolist
+        :return: Return content view
+        """
+
+        content_view = entities.ContentView(organization=org).create()
+        content_view.repository = repolist
+        content_view = content_view.update(['repository'])
+        call_entity_method_with_timeout(content_view.publish, timeout=3400)
+        content_view = content_view.read()
+        return content_view
+
+    def cleanup_of_provisioned_server(self, hostname=None, provisioning_server=None,
+                                      distro=None):
+        """ Cleanup the VM from provisioning server
+
+        :param: str hostname: The content host hostname
+        :param: str provisioning_server: provision server name
+        :param: str distro: distro type
+        """
+        if hostname:
+            vm = VirtualMachine(
+                hostname=hostname,
+                target_image=hostname,
+                provisioning_server=provisioning_server,
+                distro=distro,
+            )
+            vm._created = True
+            vm.destroy()
+
+    def assertIn(self, member, container):
```

Comment:
unittest already provides that function. The fact that you need to copy it to your class is sign of deeper problems with this approach, for me.

Either force tests to run these assertions themselves, or just use `assert` in this file.

Reply:
Agree, I think no need of assertIn, We can evaluate it by simple assert statement.